# Calculate format=diff merge(gnome-base/gnome-control-center)==46.4-r2,47.7
From 9c84b6a6939061173468cb4e5e4be5e469571f4a Mon Sep 17 00:00:00 2001
From: nemofsan <nemofsan@yandex.ru>
Date: Sat, 19 Oct 2024 13:56:06 +0300
Subject: [PATCH] added: cl-console-gui-update for calculate linux in
 info-overvie

---
 panels/system/cc-system-panel.c | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/panels/system/cc-system-panel.c b/panels/system/cc-system-panel.c
index 36f1b8190..7c81577b8 100644
--- a/panels/system/cc-system-panel.c
+++ b/panels/system/cc-system-panel.c
@@ -88,11 +88,18 @@ gpk_update_viewer_exists (void)
   return path != NULL;
 }
 
+static gboolean
+cl_console_gui_update_exist (void)
+{
+  g_autofree gchar *path = g_find_program_in_path ("cl-console-gui-update");
+  return path != NULL;
+}
+
 static gboolean
 show_software_updates_group (CcSystemPanel *self)
 {
   return (gnome_software_exists () && gnome_software_allows_updates ()) ||
-         gpk_update_viewer_exists ();
+         gpk_update_viewer_exists () || cl_console_gui_update_exist ();
 }
 
 static void
@@ -102,7 +109,12 @@ cc_system_page_open_software_update (CcSystemPanel *self)
   gboolean ret;
   char *argv[3];
 
-  if (gnome_software_exists ())
+  if (cl_console_gui_update_exist ())
+    {
+      argv[0] = "cl-console-gui-update";
+      argv[1] = NULL;
+    }
+  else if (gnome_software_exists ())
     {
       argv[0] = "gnome-software";
       argv[1] = "--mode=updates";
-- 
2.45.2

